### DISCUSSION TITLE
How to specify fields returned with token when authenticating with authWithOAuth2Code

### DISCUSSION THREAD

**Discussion by laszlo1337** [pocketbase/pocketbase#3868] [Q&A] [answered]
Hello. 

Using `authWithPassword` I can pass options and define which fields I want to be returned with the encoded token, which is then saved to cookies. 

How do I specify the same using `authWithOAuth2Code` method?

**Comment by ganigeorgiev** [comment, marked as answer]
If you use JS SDK v0.17.0+ it should be possible to pass as last options argument the `fields` query param:

```js
await pb.collection("users").authWithOAuth2Code(
   provider,
   code,
   codeVerifier,
   redirectUrl,
   {}, // additional user create data
   {
      fields: "..."
   },
)
```
